backend/app/api/routes/podcasts.py

```diff
@@ -0,0 +1,151 @@
+import os
+import uuid
+from typing import Any, Optional, List
+
+from fastapi import APIRouter, HTTPException
+from fastapi.responses import StreamingResponse, JSONResponse
+from pydantic import BaseModel
+
+from app.api.deps import CurrentUser, SessionDep
+from sqlmodel import select
+from sqlalchemy.orm import selectinload
+from app.core.config import settings
+from app.models.podcast import Podcast
+from app.schemas.public import PodcastPublic, PodcastsPublic
+from app.services.podcast_service import generate_podcast_for_course
+
+router = APIRouter(prefix="/podcasts", tags=["podcasts"])
+
+
+@router.get("/{course_id}", response_model=PodcastsPublic)
+def list_podcasts(course_id: uuid.UUID, session: SessionDep, current_user: CurrentUser) -> Any:
+    pods = session.exec(select(Podcast).where(Podcast.course_id == course_id)).all()
+    return PodcastsPublic(data=[PodcastPublic.model_validate(p) for p in pods])
+
+
+class GeneratePodcastRequest(BaseModel):
```

Comment:
The class belongs to the schemas folder if it is a Pydantic model or the models folder if it is a DB table.